src/platform_impl/windows/event_loop.rs

```diff
@@ -1937,6 +2037,49 @@ unsafe extern "system" fn thread_event_target_callback<T: 'static>(
             function();
             0
         }
+        _ if msg == *PROCESS_NEW_EVENTS_MSG_ID => {
+            winuser::PostThreadMessageW(
+                subclass_input.event_loop_runner.wait_thread_id(),
+                *CANCEL_WAIT_UNTIL_MSG_ID,
+                0,
+                0,
+            );
+
+            // if the control_flow is WaitUntil, make sure the given moment has actually passed
+            // before emitting NewEvents
+            if let ControlFlow::WaitUntil(wait_until) =
+                subclass_input.event_loop_runner.control_flow()
+            {
+                let mut msg = mem::zeroed();
+                while Instant::now() < wait_until {
+                    if 0 != winuser::PeekMessageW(&mut msg, ptr::null_mut(), 0, 0, 0) {
+                        // This works around a bug in PeekMessageW. If the message PeekMessageW
```

Comment:
This is not a bug in `PeekMessageW` but a documented feature.

Reply:
Ah, right, I almost forgot - the Windows API has no bugs. Every exhibited behavior is a feature and will remain consistent until the end of time.

Reply:
Some of those APIs probably date back from Windows 3.1.